uint8[25] SAMMode
uint16[25] SAMPos12
